prowler/providers/kubernetes/kubernetes_provider.py

```diff
@@ -0,0 +1,72 @@
+import os
+import sys
+
+from kubernetes import client, config
+
+from prowler.lib.logger import logger
+
+
+class Kubernetes_Provider:
+    def __init__(
+        self,
+        kubeconfig_file: str,
+        context: list,
+    ):
+        logger.info("Instantiating Kubernetes Provider ...")
+        self.api_client, self.context = self.__set_credentials__(
+            kubeconfig_file, context
+        )
+        if not self.api_client:
+            logger.critical("Failed to set up a Kubernetes session.")
+            sys.exit(1)
+
+    def __set_credentials__(self, kubeconfig_file, context):
+        try:
+            if kubeconfig_file:
+                # Use kubeconfig file if provided
+                config.load_kube_config(
+                    config_file=os.path.abspath(kubeconfig_file), context=context
+                )
+            else:
+                # Otherwise try to load in-cluster config
+                config.load_incluster_config()
+            context = config.list_kube_config_contexts()[0][0]
```

Comment:
Why this `[0][0]`? is it possible to recover it in a more explicit way?

Reply:
No, it is always the same structure and it has to be parsed in that way.